coding.md

```diff
@@ -0,0 +1,109 @@
+# Ursula Developer Coding Standards
+In the beginning, there were no guidelines. And it was good. But that
+didn't last long. As more and more people added more and more code,
+we realized that we needed a set of coding standards to make sure that
+Ursula at least *attempted* to display some form of consistency.
+
+In addition, newer versions of Ansible will change what format these
+files can be, so we are being proactive with a coding style to prevent
+having to clean up more things when we move to new versions of Ansible.
+
+Thus, these coding standards/guidelines were developed. Note that not
+all of Ursula adheres to these standards just yet. Some older code has
+just not been changed yet. But be clear, all new code *must* adhere to
+these guidelines.
+
+Below are the patterns that we expect Ursula developers to follow.
+
+## General
+* Our chosen file extension for YAML files is `.yml`.
+
+* Make every attempt to keep lines under 80 chars long.
+
+## Variables
+* When referencing variables with mustache brackets, use spaces around the
+  brackets: `"{{ a_var }}"`. 
+
+* Do not use "raw variables" in `with_items`. Instead quote them:
+  `with_items: "{{ nova.list_var }}"`
+
+* Use descriptive varibles when setting facts or registering output. Avoid
+  using the same variable name used elsewhere within Ursula.
+
+* Boolean variables should always be ran through the `|bool` filter to
+  normalize the input into a Boolean.
+
+* Do not rely on variable settings in an example env defaults file. These
+  files are just for example and may not be used in production. Instead rely
+  on role defaults or the `|default()` filter to set a default value.
+
+* Filter application should be done without spaces between the variable,
+  the pipe symbol, and the filter name: `{{ foo|default(False)|bool }}`
```

Comment:
I also agree *with spaces* is actually easier to read and scan across than without them.